tests/st/test_ipip.py

```diff
@@ -29,3 +32,56 @@ class TestIPIP(TestBase):
     @skip("Not written yet")
     def test_ipip(self):
         pass
+
+    def test_ipip_addr_assigned(self):
+        with DockerHost('host', dind=False, start_calico=False) as host:
+            # Set up first pool before Node is started, to ensure we get tunl IP on boot
+            ipv4_pool = "10.0.1.0/24"
+            host.calicoctl("pool add %s --ipip" % ipv4_pool)
+            host.start_calico_node()
+            self.assert_tunl_ip(host, r'inet 10\.0\.1\.[\d]{1,3}/32', expect=True)
+
+            # Test that removing tunl removes the tunl IP.
+            host.calicoctl("pool remove %s" % ipv4_pool)
+            self.assert_tunl_ip(host, r'inet 10\.0\.1\.[\d]{1,3}/32', expect=False)
+
+            # Test that readding the pool triggers the confd watch and we get an IP
+            host.calicoctl("pool add %s --ipip" % ipv4_pool)
+
+            # Test we have a tunl interface assigned from startup
+            self.assert_tunl_ip(host, r'inet 10\.0\.1\.[\d]{1,3}/32', expect=True)
+
+            # Test that by adding another pool, then deleting the first,
+            # we remove the original IP, and allocate a new one from the new pool
+            new_ipv4_pool = "192.168.0.0/16"
+            host.calicoctl("pool add %s --ipip" % new_ipv4_pool)
+            host.calicoctl("pool remove %s" % ipv4_pool)
+            self.assert_tunl_ip(host, r'inet 192\.168\.[\d]{1,3}.[\d]{1,3}/32')
+
+    def assert_tunl_ip(self, host, re_expr, expect=True):
```

Comment:
This is a rather odd interface IMHO.  You are asserting the tunl IP but you are passing in the net regex.  This function should really be doing the match and pulling out the IP address - and then I'd imagine you'd want to check that the IP is from one a number of available pools or CIDRs.